src/SFML/System/String.cpp

```diff
@@ -37,6 +37,43 @@
 
 namespace sf
 {
+////////////////////////////////////////////////////////////
```

Comment:
SFML tries hard to keep headers minimal so I implemented these functions in the source file. I’m open to reevaluating that if there are convincing benchmarks to the contrary. 

Because this header is used in many places there are compile time benefits to not inlining the functions.

Reply:
It's not a benchmark, but maybe it's convincing enough: https://godbolt.org/z/4Mb3Mxhcs. Also note that these functions are really small and thus very cheap to instantiate.